src/Apps/Order/Routes/__tests__/Shipping.jest.tsx

```diff
@@ -192,346 +231,435 @@ describe("Shipping", () => {
 
-      fillAddressForm(page.root, {
+      const address = {
         ...validAddress,
         region: "New Brunswick",
         country: "US",
-      })
+      }
+      fillAddressForm(page.root, address)
 
       await page.clickSubmit()
-      expect(mutations.mockFetch.mock.calls.map(call => call[1].input))
-        .toMatchInlineSnapshot(`
-        Array [
-          Object {
-            "fulfillmentType": "SHIP",
-            "id": "1234",
-            "phoneNumber": "8475937743",
-            "shipping": Object {
-              "addressLine1": "14 Gower's Walk",
-              "addressLine2": "Suite 2.5, The Loom",
-              "city": "Whitechapel",
-              "country": "US",
-              "name": "Artsy UK Ltd",
-              "phoneNumber": "",
-              "postalCode": "E1 8PY",
-              "region": "New Brunswick",
+      expect(mockCommitMutation).toHaveBeenCalledTimes(2)
+      expect(mockCommitMutation).toHaveBeenNthCalledWith(
+        1,
+        expect.objectContaining({
+          variables: {
+            input: {
+              id: "1234",
+              fulfillmentType: "SHIP",
+              phoneNumber: address.phoneNumber,
+              shipping: {
+                ...address,
+                phoneNumber: "",
+              },
             },
           },
-          Object {
-            "attributes": Object {
-              "addressLine1": "14 Gower's Walk",
-              "addressLine2": "Suite 2.5, The Loom",
-              "city": "Whitechapel",
-              "country": "US",
-              "name": "Artsy UK Ltd",
-              "phoneNumber": "8475937743",
-              "postalCode": "E1 8PY",
-              "region": "New Brunswick",
+        })
+      )
+      expect(mockCommitMutation).toHaveBeenNthCalledWith(
+        2,
+        expect.arrayContaining([
+          expect.anything(),
+          expect.objectContaining({
+            variables: {
+              input: {
+                attributes: address,
+              },
             },
-          },
-        ]
-      `)
+          }),
+        ])
+      )
     })
 
     it("commits the mutation with pickup option", async () => {
-      const page = await buildPage()
+      mockCommitMutation.mockResolvedValueOnce(settingOrderShipmentSuccess)
+      const wrapper = getWrapper({
+        CommerceOrder: () => testOrder,
+        Me: () => emptyTestMe,
+      })
+      const page = new ShippingTestPage(wrapper)
+
       await page.selectPickupOption()
       fillInPhoneNumber(page.root, { isPickup: true, value: "2813308004" })
-      expect(mutations.mockFetch).not.toHaveBeenCalled()
+      expect(mockCommitMutation).not.toHaveBeenCalled()
+
       await page.clickSubmit()
-      expect(mutations.mockFetch).toHaveBeenCalledTimes(1)
-      expect(mutations.lastFetchVariables.input.fulfillmentType).toBe("PICKUP")
+      expect(mockCommitMutation).toHaveBeenCalledTimes(1)
+      expect(mockCommitMutation).toHaveBeenCalledWith(
```

Comment:
since we don't have the custom variables to watch we only check if the mocked mutation is being called with the right values